homeassistant/components/device_tracker/__init__.py

```diff
@@ -196,86 +208,93 @@ def __init__(self, hass: HomeAssistantType, consider_home: timedelta,
                                 dev.mac)
         self.consider_home = consider_home
         self.track_new = track_new
-        self.lock = threading.Lock()
-
-        for device in devices:
-            if device.track:
-                device.update_ha_state()
-
         self.group = None  # type: group.Group
 
     def see(self, mac: str=None, dev_id: str=None, host_name: str=None,
             location_name: str=None, gps: GPSType=None, gps_accuracy=None,
             battery: str=None, attributes: dict=None):
         """Notify the device tracker that you see a device."""
-        with self.lock:
-            if mac is None and dev_id is None:
-                raise HomeAssistantError('Neither mac or device id passed in')
-            elif mac is not None:
-                mac = str(mac).upper()
-                device = self.mac_to_dev.get(mac)
-                if not device:
-                    dev_id = util.slugify(host_name or '') or util.slugify(mac)
-            else:
-                dev_id = cv.slug(str(dev_id).lower())
-                device = self.devices.get(dev_id)
-
-            if device:
-                device.seen(host_name, location_name, gps, gps_accuracy,
-                            battery, attributes)
-                if device.track:
-                    device.update_ha_state()
-                return
+        run_coroutine_threadsafe(
```

Comment:
I don't think that we have to wait for the result, so fire_coroutine_threadsafe should be ok.